cmd/kube-proxy/app/server.go

```diff
@@ -1114,3 +1120,30 @@ func getNodeIPs(ctx context.Context, client clientset.Interface, name string) []
 	}
 	return nodeIPs
 }
+
+// waitUntilNodeCache waits until the node with the provided name is created in the server side.
+func waitUntilNodeCache(ctx context.Context, client clientset.Interface, name string) error {
```

Comment:
we have another watch here https://github.com/kubernetes/kubernetes/blob/be4535bd34ac09e57a0b1c478cbcebd051a42598/cmd/kube-proxy/app/server_linux.go#L455


we run the informers here https://github.com/kubernetes/kubernetes/blob/be4535bd34ac09e57a0b1c478cbcebd051a42598/cmd/kube-proxy/app/server.go#L988-L1011

Since IIRC kube-proxy will not work without the informers  synced

1. we need to first wait until the informers cache are synced

```
	if !cache.WaitForNamedCacheSync("kube-proxy, ctx.Done(), c.nodeInfomerSynced, c.serviceInformerSynced, c.endpointSliceSyncer) {
		return
	}
```

2. replace the watches and Get() calls to use the informer lister